File: java8/src/main/java/alvin/java8/MainActivity.java
package alvin.java8;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private Button btnClickMe;
    private TextView textResult;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.textResult = (TextView) findViewById(R.id.text_result);

        this.btnClickMe = (Button) findViewById(R.id.btn_click_me);
        this.btnClickMe.setOnClickListener(view -> this.textResult.setText(R.string.text_hello));
    }
}

File: ui.actionbar/src/main/java/alvin/ui/actionbar/main/MainActivity.kt
package alvin.ui.actionbar.main

import alvin.ui.actionbar.R
import alvin.ui.actionbar.simple.SimpleActionBarActivity
import alvin.ui.actionbar.toolbar.ToolbarActivity
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.intentFor

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        setupButtons()
    }

    private fun setupButtons() {
        arrayOf(btn_simple_action, btn_action_home_up).forEach {
            it.setOnClickListener { b ->
                startActivity(when (b.id) {
                    btn_simple_action.id -> intentFor<SimpleActionBarActivity>()
                    btn_action_home_up.id -> intentFor<ToolbarActivity>()
                    else -> throw IllegalArgumentException()
                })
            }
        }
    }
}
